matcher/match.go

```diff
@@ -85,19 +86,40 @@ func matchRectangular(pos position.Position, filter *repository.Filter) (bool, e
 func matchRound(pos position.Position, filter *repository.Filter) (bool, error) {
 	var rfilter repository.RoundFilter
+	var limit float64 = 360
 	if err := json.Unmarshal(filter.Configuration, &rfilter); err != nil {
 		return false, err
 	}
-
-	matched := (pos.Latitude-rfilter.CenterLatitude)*(pos.Latitude-rfilter.CenterLatitude)+
-		(pos.Longitude-rfilter.CentreLongitude)*(pos.Longitude-rfilter.CentreLongitude) <=
-		(rfilter.Radius * rfilter.Radius)
-
-	return xor(matched, filter.Reversed), nil
+	switch {
+	//round filter longitude and lalilude in critical position(near to +-180)
+	case (limit/2-math.Abs(float64(rfilter.CenterLatitude))-float64(rfilter.Radius)) <= 0 && (limit/2-math.Abs(float64(rfilter.CentreLongitude))-float64(rfilter.Radius)) <= 0:
+		matched := (limit-math.Abs(float64(pos.Latitude))-math.Abs(float64(rfilter.CenterLatitude)))*(limit-math.Abs(float64(pos.Latitude))-math.Abs(float64(rfilter.CenterLatitude)))+
+			(limit-math.Abs(float64(pos.Longitude))-math.Abs(float64(rfilter.CentreLongitude)))*(limit-math.Abs(float64(pos.Longitude))-math.Abs(float64(rfilter.CentreLongitude))) <=
+			float64(rfilter.Radius*rfilter.Radius)
+		return xor(matched, filter.Reversed), nil
+	//round filter lalilude in critical position(near to +-180)
+	case (limit/2-math.Abs(float64(rfilter.CenterLatitude))-float64(rfilter.Radius)) <= 0 && !((limit/2 - math.Abs(float64(rfilter.CentreLongitude)) - float64(rfilter.Radius)) <= 0):
+		matched := (limit-math.Abs(float64(pos.Latitude))-math.Abs(float64(rfilter.CenterLatitude)))*(limit-math.Abs(float64(pos.Latitude))-math.Abs(float64(rfilter.CenterLatitude)))+
+			float64((pos.Longitude-rfilter.CentreLongitude)*(pos.Longitude-rfilter.CentreLongitude)) <=
+			float64(rfilter.Radius*rfilter.Radius)
+		return xor(matched, filter.Reversed), nil
+	//round filter longitude in critical position(near to +-180)
+	case !((limit/2 - math.Abs(float64(rfilter.CenterLatitude)) - float64(rfilter.Radius)) <= 0) && (limit/2-math.Abs(float64(rfilter.CentreLongitude))-float64(rfilter.Radius)) <= 0:
+		matched := float64((pos.Latitude-rfilter.CenterLatitude)*(pos.Latitude-rfilter.CenterLatitude))+
+			(limit-math.Abs(float64(pos.Longitude))-math.Abs(float64(rfilter.CentreLongitude)))*(limit-math.Abs(float64(pos.Longitude))-math.Abs(float64(rfilter.CentreLongitude))) <=
+			float64(rfilter.Radius*rfilter.Radius)
+		return xor(matched, filter.Reversed), nil
+	//round filter canter in safe position (not near to 180)
+	case !(limit/2-math.Abs(float64(rfilter.CenterLatitude))-float64(rfilter.Radius) <= 0) && !((limit/2 - math.Abs(float64(rfilter.CentreLongitude)) - float64(rfilter.Radius)) <= 0):
+		matched := (pos.Latitude-rfilter.CenterLatitude)*(pos.Latitude-rfilter.CenterLatitude)+
+			(pos.Longitude-rfilter.CentreLongitude)*(pos.Longitude-rfilter.CentreLongitude) <=
+			(rfilter.Radius * rfilter.Radius)
+		return xor(matched, filter.Reversed), nil
+	default:
+		return false, nil
+	}
```

Comment:
Could you please explain how does it work?

Reply:
we calculate default vector module. if it less than radius we return true(or reversed), if false we compare coordinats with limit 180 and radius. so  if 180 - point- ordinate <=0 we calculated vector part for this ordinate with inverted logic 360-centerOrdinate-posOrdinate and in this case we return true of point is in the radius event in lositions like 179 0 and -179 0.  -179 0 with radius 10(so we are in critical position) and point -169